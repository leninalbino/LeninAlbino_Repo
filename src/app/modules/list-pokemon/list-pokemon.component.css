@import url('https://fonts.googleapis.com/css2?family=Raleway:ital,wght@1,200&display=swap');

*{
  margin: 0;
  padding: 0;
  box-sizing: border-box; /*para simplificar el calcular de las cajas*/
}

main{
  background: #1b2831; /**/
  font-family: "Raleway", sans-serif; /*Fuente de texto*/
}

.headers{
  display: grid;
  align-items: center;
  text-align: center;
  height: 30px;
  width: 50%;
  padding: 4rem;
  /* padding-bottom: 0px; */
  font-size: large;
  font-family: "Raleway", sans-serif; /*Fuente de texto*/
}
input{
  height: 30px;
}
.container{
  display: flex ; /**/
}

.contents{
  display: grid;
  grid-template-columns: repeat(2,1fr); /*mostrar cuantas columnas los elementos*/
  align-items: center;
  width: 50%;
  padding: 4rem;
}
.elements{
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 2rem;
}
.main-image{
  position: relative;
  transition: all 0.04s;
  display: flex;
  justify-content: center;
  align-items: center;
}
.elements:hover{
  background-color: #273b49;
  border-radius: 10px;
}
.main-image:hover{
  width: 100%;
  transform: scale(1.8);
}
h2{
  color: #ffffff;
  font-family: "Raleway", sans-serif; /*Fuente de texto*/
}
p{
  color: #ffffff;
  text-align: center;
  margin-top: 1rem;
  line-height: 2;
  font-size: 18px;
}

.buttons{
  display: flex;
  justify-content: space-around;
  width: 50%;
  padding: 4rem;

}
button{
  height: 35px;
  width: 95px;
  padding: 0%;
  font-size: 20px;
  border: none;
  outline: none;
  border-radius: 5px;
  overflow: hidden;
  font-weight: 500;
  background: #009578;
  color: #ffffff;
  font-family: "Raleway", sans-serif;
}
.container .details{
  display: grid;
  width: 50%;
  padding: 4rem;

}
h3{
  color: #ffffff;
  text-align: center;
  font-family: "Raleway", sans-serif; /*Fuente de texto*/
}

span{
  font-family: "Raleway", sans-serif; /*Fuente de texto*/
  color: #ffffff;
}
img{
  display: flex;
}
