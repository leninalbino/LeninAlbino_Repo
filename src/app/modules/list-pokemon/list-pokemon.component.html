<!-- <main>
  <form [formGroup]="form">
    <h1>Listado de Pokemon</h1>
    <input formControlName="search" (keyup)="searchPokemon()" matInput placeholder="search">
    <div class="container">

      <section class="content">
        <article *ngFor="let pokemon of last_list_pokemons | pagesPokemon:page" (click)="getOneById(pokemon[0].id)"  >
          <img [src]="pokemon[0].image">
          <h2 >{{pokemon[0].id}}</h2>
          <h2>{{pokemon[0].name}}</h2>
        </article>
      </section>
      <div class="descript">
        <article class="detail" *ngIf="view">
          <img height="200px" [src]="detailPokemon.sprites.front_default">
          <h2> # {{detailPokemon.id}} </h2>
          <h2>{{detailPokemon.name}}</h2>
          <div>
            <h3>Types:</h3>
            <span *ngFor="let type of detailPokemon.types">
              {{type.type.name}}
            </span>
            <h3>Peso</h3>
            <p> {{detailPokemon.weight}} kg </p>
            <span class="image">
              <img class="img"  [src]="detailPokemon.sprites.front_default">
            </span>
            <span *ngIf="detailPokemon.sprites.front_female" class="image">
              <img class="img" [src]="detailPokemon.sprites.front_female">
            </span>
            <div *ngIf="detailPokemon.sprites.front_shiny" class="image">
              <img class="img" [src]="detailPokemon.sprites.front_shiny">
            </div>
            <div *ngIf="detailPokemon.sprites.front_shiny_female" class="image">
              <img class="img" [src]="detailPokemon.sprites.front_shiny_female">
            </div>
              <div *ngIf="detailPokemon.sprites.back_default" class="image">
              <img class="img" [src]="detailPokemon.sprites.back_default">
            </div>
              <div *ngIf="detailPokemon.sprites.back_female" class="image">
              <img class="img" [src]="detailPokemon.sprites.back_female">
            </div>
              <div *ngIf="detailPokemon.sprites.back_shiny" class="image">
              <img class="img" [src]="detailPokemon.sprites.back_shiny">
            </div>
              <div *ngIf="detailPokemon.sprites.back_shiny_female" class="image">
              <img class="img" [src]="detailPokemon.sprites.back_shiny_female">
            </div>

            <br>
            <h3>Movimientos</h3>
            <span *ngFor="let move of detailPokemon.moves">
              {{move.move.name}}
            </span>
          </div>
        </article>

      </div>
    </div>
  </form>
  <div class="buttons">
    <button (click)=" prevPage()">Atras</button>
    <button (click)=" nextPage()">Siguiente</button>
  </div>

</main> -->
<main>
  <section class="headers">
    <h2>Listado de Pokemon</h2>
    <input #txtSearch (keyup)="searchPokemon(txtSearch.value)" matInput placeholder="search">
  </section>

  <div class="container">
    <div class="contents">

      <section class="elements" *ngFor="let pokemon of last_list_pokemons | pagesPokemon:page" (click)="getOneById(pokemon[0].id)">
        <figure class="main-image">
          <img [src]="pokemon[0].image">
        </figure>
        <h2>{{pokemon[0].id}}</h2>
        <p>{{pokemon[0].name}}</p>
      </section>

    </div>
    <div class="details" *ngIf="view">

     <section class="pokemon-detail">
      <figure class="main-image">
        <img [src]="detailPokemon.sprites.front_default">
      </figure>
      <h3 > # {{detailPokemon.id}} </h3>
      <h3>{{detailPokemon.name}}</h3>
        <h2>Types:</h2>
        <span *ngFor="let type of detailPokemon.types">
          {{type.type.name}}
        </span>
        <h2>Peso</h2>
        <h2> {{detailPokemon.weight}} kg </h2>
        <div class="image">
          <img class="img"  [src]="detailPokemon.sprites.front_default">
        </div>
        <div class="image" *ngIf="detailPokemon.sprites.front_female" class="image">
          <img class="img" [src]="detailPokemon.sprites.front_female">
        </div>
        <figure class="image" *ngIf="detailPokemon.sprites.front_shiny" class="image">
          <img class="img" [src]="detailPokemon.sprites.front_shiny">
        </figure>
        <figure class="image" *ngIf="detailPokemon.sprites.front_shiny_female" class="image">
          <img class="img" [src]="detailPokemon.sprites.front_shiny_female">
        </figure>
        <figure class="image" *ngIf="detailPokemon.sprites.back_default" class="image">
          <img class="img" [src]="detailPokemon.sprites.back_default">
        </figure>
        <figure class="image" *ngIf="detailPokemon.sprites.back_female" class="image">
          <img class="img" [src]="detailPokemon.sprites.back_female">
        </figure>
        <figure class="image" *ngIf="detailPokemon.sprites.back_shiny" class="image">
          <img class="img" [src]="detailPokemon.sprites.back_shiny">
        </figure>
        <figure class="image" *ngIf="detailPokemon.sprites.back_shiny_female" class="image">
          <img class="img" [src]="detailPokemon.sprites.back_shiny_female">
        </figure>

        <br>
        <h2>Movimientos</h2>
        <span *ngFor="let move of detailPokemon.moves">
          {{move.move.name}}
        </span>

     </section>
    </div>
  </div>
  <div class="buttons">
    <button (click)=" prevPage()">Atras</button>
    <button (click)=" nextPage()">Siguiente</button>
</div>
</main>

