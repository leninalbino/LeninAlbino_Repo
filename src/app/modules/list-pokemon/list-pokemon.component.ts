import { PokemonService } from '../services/list-pokemon.service';
import { Component, OnInit } from '@angular/core';
import { Pokemon } from '../models/pokemon';

@Component({
  selector: 'app-list-pokemon',
  templateUrl: './list-pokemon.component.html',
  styleUrls: ['./list-pokemon.component.css']
})
export class ListPokemonComponent implements OnInit {

  view: boolean = false;

  public page: number = 0;
  detailPokemon:any={};

  get_List_Pokemon_Of_Object: any[] = [];
  new_List_pokemons: any[] = [];
  pokemons: Pokemon[] = [];
  last_list_pokemons: any[] = [];

  constructor(
    private _pokemonService: PokemonService
  ) { }

  ngOnInit(): void {
    this.getPokemon();
  }

  objectKeys(objeto: any) {
    const values = Object.values(objeto);
    console.log(values); // echa un vistazo por consola para que veas lo que hace "Object.keys"
    return values;
  }
  getPokemon() {
    this._pokemonService.getPokemons().subscribe(pokedata => {
      //Todo
      console.log(pokedata);
      this.getImage(pokedata);
    })
  }

  getImage(object: any): Pokemon[] {
    this.get_List_Pokemon_Of_Object = Object.values(object.results);
    this.get_List_Pokemon_Of_Object.filter((data: any) => {
      const url = data.url.split('/');
      const id = url[6];
      const image = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`;

      this.pokemons = [
        {
          id: id,
          name: data.name,
          image: image
        }
      ]

      this.new_List_pokemons.push(this.pokemons);
      this.last_list_pokemons = this.new_List_pokemons;
      // console.log(this.last_list_pokemons);

    })
    return this.new_List_pokemons;
  }


  getOneById(id:number){
    this._pokemonService.getOneById(id).subscribe(dataDetail=>{
      this.viewDetalle();
      this.detailPokemon = dataDetail;
      console.log(this.detailPokemon);
    })
  }

  searchPokemon(value:any) {
    console.log();

    let search = value.trim().toUpperCase()
    if (value.trim().length >= 1) {
      let pokemon = this.new_List_pokemons.filter(e => e[0].id == value || e[0].name.toUpperCase()
                                                                                      .toLowerCase()
                                                                                      .includes(value.toUpperCase().toLowerCase()));
      this.last_list_pokemons = pokemon;
    } else {
      this.last_list_pokemons = this.new_List_pokemons;
    }
  }


  nextPage() {
    this.page += 8;
  }

  prevPage() {
    if ( this.page > 0 )
      this.page -= 8;
  }
  viewDetalle() {

    this.view = true;
  }

}



