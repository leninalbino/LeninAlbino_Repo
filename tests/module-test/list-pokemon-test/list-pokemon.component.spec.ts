import { Pokemon } from './../../../src/app/modules/models/pokemon';
import { PagesPokemonPipe } from './../../../src/app/modules/pipes/pages-pokemon.pipe';
import { PokemonService } from './../../../src/app/modules/services/list-pokemon.service';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { ListPokemonComponent } from './../../../src/app/modules/list-pokemon/list-pokemon.component';
import { ComponentFixture, TestBed } from '@angular/core/testing';
describe('ListPokemonComponent', () => {
  let component: ListPokemonComponent;
  let fixture: ComponentFixture<ListPokemonComponent>;

  let compiled: HTMLElement; //porbar el Htlm
  let service: PokemonService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ ListPokemonComponent,
                      PagesPokemonPipe ],

      imports: [HttpClientTestingModule],
      providers: [PokemonService]
    })
    .compileComponents();

    fixture = TestBed.createComponent(ListPokemonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    //llenamos los valores
    compiled = fixture.nativeElement;
    service = TestBed.inject(PokemonService);
    httpMock = TestBed.inject(HttpTestingController)
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('must match the snapshot',() =>{
    expect(compiled).toMatchSnapshot();
  });

  test('clicking the button should increment by 8', ()  =>{
    const buttons = compiled.querySelectorAll('button');
    buttons[1].click();
    expect(component.page).toBe(8)
  })

  test('must show the main title', ()  =>{
    const h2 = compiled.querySelector('h2');
    expect(h2?.textContent).toContain('Listado de Pokemon')
  });

  test('must load the pokemon', ()  =>{
    const dummyPokemons = [
      {
        id: '1',
        name:'bulbasaur',
        image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
      }
    ]
    //httpMock, es un ojo o un observador de http que esta en el servicio
    //console.log(dummyPokemons);
    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/?limit=1500');
    expect(request.request.method).toBe('GET');// probamos que el tipo de respuesta sea Get
    request.flush(dummyPokemons)// para ver si ese request regreso de tipo dumyPokemon

    fixture.detectChanges();//detectar los cambios
  })

});
