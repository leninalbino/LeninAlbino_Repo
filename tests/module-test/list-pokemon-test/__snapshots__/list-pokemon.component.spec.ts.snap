// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPokemonComponent must match the snapshot 1`] = `
<div
  id="root1"
>
  <main>
    <section
      class="headers"
    >
      <h2>
        Listado de Pokemon
      </h2>
      <input
        matinput=""
        placeholder="search"
      />
    </section>
    <div
      class="container"
    >
      <div
        class="contents"
      >
        
      </div>
      
    </div>
    <div
      class="buttons"
    >
      <button>
        Atras
      </button>
      <button>
        Siguiente
      </button>
    </div>
  </main>
</div>
`;
